test/GoogleAnalyticsEventProcessor/constructor_test.js

```diff
@@ -0,0 +1,89 @@
+goog.module('measurementLibrary.eventProcessor.testing.GoogleAnalyticsEventProcessor.processEvent');
+goog.setTestOnly();
+
+const GoogleAnalyticsEventProcessor = goog.require('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+describe('The `constructor` of GoogleAnalyticsEventProcessor', () => {
+  let eventProcessor;
+
+  const defaultAutomaticParams = {
+    'page_path': true,
+    'page_location': true,
+    'page_title': true,
+    'user_id': true,
+    'client_id': true,
+  };
+
+  it('does not have a default arguments for ' +
+      '`api_secret` or `measurement_id`', () => {
+    eventProcessor = new GoogleAnalyticsEventProcessor();
+
+    expect(eventProcessor.apiSecret_).toBeUndefined();
+    expect(eventProcessor.measurementId_).toBeUndefined();
+  });
+
+  it('has the correct default arguments for `measurement_url`, ' +
+  '`client_id_expires`, and `automatic_params`', () => {
+    eventProcessor = new GoogleAnalyticsEventProcessor();
+
+    expect(eventProcessor.measurementUrl_).toBe('https://www.google-analytics.com/mp/collect');
+    expect(eventProcessor.clientIdExpires_).toBePositiveInfinity();
+
+    for (const key in defaultAutomaticParams) {
+      expect(eventProcessor.automaticParams_.hasOwnProperty(key)).toBeTrue();
```

Comment:
I don't think it's possible for this check to fail, unless something has messed with Object.prototype.